tests/v2/validation/pipeline/rancherversion/main.go

```diff
@@ -0,0 +1,357 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+
+	"os"
+	"sort"
+
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type RancherK8sVersions struct {
+	RancherImageTag             string `json:"rancherImageTag" yaml:"rancherImageTag"`
+	RancherImageTagToUpgrade    string `json:"rancherImageTagToUpgrade" yaml:"rancherImageTagToUpgrade"`
+	RancherVersion              string `json:"rancherVersion" yaml:"rancherVersion"`
+	RancherVersionToUpgrade     string `json:"rancherVersionToUpgrade" yaml:"rancherVersionToUpgrade"`
+	RancherRKE2Version          string `json:"rancherRKE2Version" yaml:"rancherRKE2Version"`
+	RancherRKE2VersionToUpgrade string `json:"rancherRKE2VersionToUpgrade" yaml:"rancherRKE2VersionToUpgrade"`
+	RancherK3sVersion           string `json:"rancherK3sVersion" yaml:"rancherK3sVersion"`
+	RancherK3sVersionToUpgrade  string `json:"rancherK3sVersionToUpgrade" yaml:"rancherK3sVersionToUpgrade"`
+	RancherRKEVersion           string `json:"rancherRKEVersion" yaml:"rancherRKEVersion"`
+	RancherRKEVersionToUpgrade  string `json:"rancherRKEVersionToUpgrade" yaml:"rancherRKEVersionToUpgrade"`
+}
+
+var (
+	rancherVersion            = os.Getenv("RANCHER_VERSION")
+	rancherVersionToUpgrade   = os.Getenv("RANCHER_VERSION_TO_UPGRADE")
+	communityBaseURL          = "https://github.com/rancher/rancher/releases/download/" + rancherVersion + "/rancher-images.txt"
+	communityBaseURLToUpgrade = "https://github.com/rancher/rancher/releases/download/" + rancherVersionToUpgrade + "/rancher-images.txt"
+	primeRelease              = true
+	primeReleaseToUpgrade     = false
+)
```

Comment:
I'm not a big fan of global variables as they make the code hard to follow because they seem to be modified as the code is ran.	I'm specifically looking at rancherVersionToUpgrade as used in the extract versions functions.

Reply:
agreed.

Reply:
rancherVersionToUpgrade - for rancher ha upgrade job, we send in a rancher version to upgrade and obtain the k8s versions based on that. that is why we need this global variable. for another scenario - local upgrade checks its going to be empty and therefore we would need to obtain the latest version and hence modified as the code is ran.

Reply:
Can you just move this inside the function it is used in?

Reply:
hmm, I'm not sure I understand. do you want me to move to the functions? what I'm trying to say is, the values for these are obtained from environment variables - Jenkins parameters.
```
rancherVersion            = os.Getenv("RANCHER_VERSION")
rancherVersionToUpgrade   = os.Getenv("RANCHER_VERSION_TO_UPGRADE")
```

Reply:
I'm just trying to figure out how to eliminate the global variables. Can you call `os.Getenv("RANCHER_VERSION_TO_UPGRADE")` when it is needed inside of main?

Also, can the bool variables be changed to constants?

Reply:
I see. I understand what you're saying. rancherVersion and rancherVersionToUpgrade are used in multiple functions. If I move it to main, I would need to send this value in multiple function call parameters and I didn't want that to be crowded.

bool variables cant be changed to constants because their values will be changed in further functions.

But what I can do is, have rancherVersion and rancherVersionToUpgrade in var and move the bool variables and communityBaseURLToUpgrade/communityBaseURL to the main function.